src/inmanta/server/protocol.py

```diff
@@ -161,13 +162,16 @@ def __init__(self, name):
         self._handlers = []
         self._sched = Scheduler()
 
+    @gen.coroutine
     def prestart(self, server: RESTServer):
         """Called by the RestServer host prior to start, can be used to collect references to other server slices"""
         pass
 
+    @gen.coroutine
     def start(self):
         pass
 
+    @gen.coroutine
```

Comment:
please document that fact that 
1- stop is blocking
2- there is at this point no way to indicate the startup/shutdown order